jupyterhub_traefik_proxy/etcd.py

```diff
@@ -261,7 +202,7 @@ def _stop_traefik(self):
         try:
             # Check if traefik was launched
             pid = self.traefik_process.pid
```

Comment:
We shouldn't be ckecing for traefik_process.pid if `self.should_start`. Most of the time, the traefik process will actually not be a subprocess managed by `self.start()`, it will already be running before we launch here.